extractor/filesystem/osv/metadata.go

```diff
@@ -21,3 +21,27 @@ type Metadata struct {
 	Ecosystem string
 	CompareAs string
 }
+
+// DepGroups provides access to the dependency groups property in some metadata objects
+type DepGroups interface {
+	DepGroups() []string
+}
+
+// osv.DepGroupMetadata is a metadata struct that only supports DepGroups
+type DepGroupMetadata struct {
+	DepGroupVals []string
+}
+
+var _ DepGroups = DepGroupMetadata{}
+
+// DepGroups return the dependency groups property in the metadata
+func (dgm DepGroupMetadata) DepGroups() []string {
+	return dgm.DepGroupVals
+}
+
+// DistroVersionMetadata contains distro versions
```

Comment:
Unused at the moment so let's introduce it in a follow-up PR with the other extractors.